test/cppsuite/tests/background_compact.cpp

```diff
@@ -0,0 +1,328 @@
+                /* Start a transaction if possible. */
+                tw->txn.try_begin();
+
+                /* Choose a random key to delete. */
+                int ret = rnd_cursor->next(rnd_cursor.get());
+
+                if (ret != 0) {
+                    /*
+                     * It is possible not to find anything if the collection is empty. In that case,
+                     * finish the current transaction as we might be able to see new records after
+                     * starting a new one.
+                     */
+                    if (ret == WT_NOTFOUND) {
+                        WT_IGNORE_RET_BOOL(tw->txn.commit());
+                    } else if (ret == WT_ROLLBACK) {
+                        tw->txn.rollback();
+                    } else {
+                        testutil_die(ret, "Unexpected error returned from cursor->next()");
+                    }
+                    testutil_check(rnd_cursor->reset(rnd_cursor.get()));
+                    break;
+                }
+
+                const char *key_str;
+                testutil_check(rnd_cursor->get_key(rnd_cursor.get(), &key_str));
+
+                std::string first_key = key_str;
+                uint64_t truncate_range =
+                  random_generator::instance().generate_integer<uint64_t>(0, 100);
+                std::string end_key = tw->pad_string(
+                  std::to_string(std::stoi(first_key) + truncate_range), first_key.size());
+
+                /*
+                 * If we generate an invalid range or our truncate fails rollback the transaction.
+                 */
+                if (end_key == first_key || !tw->truncate(coll.id, first_key, end_key, "")) {
+                    tw->txn.rollback();
+                    logger::log_msg(LOG_TRACE, log_prefix + "truncate failed");
+                    continue;
+                }
+
+                if (tw->txn.commit()) {
+                    logger::log_msg(LOG_TRACE,
+                      log_prefix + " committed truncation of " + std::to_string(truncate_range) +
+                        " records.");
+                    keys_truncated += truncate_range;
+                } else
+                    logger::log_msg(LOG_TRACE,
+                      log_prefix + "failed to commit truncation of " +
+                        std::to_string(truncate_range) + " records.");
+
+                /* Reset our cursor to avoid pinning content. */
+                testutil_check(rnd_cursor->reset(rnd_cursor.get()));
+            }
+
+            logger::log_msg(LOG_TRACE,
+              log_prefix + "truncated " + std::to_string(keys_truncated) + " keys out of " +
+                std::to_string(n_keys_to_truncate));
+
+            /*
+             * Take a checkpoint here so we can read the correct statistics next time we hit this
+             * file.
+             */
+            testutil_check(tw->session->checkpoint(tw->session.get(), nullptr));
```

Comment:
Would it be worth it to enable the checkpoint manager instead with a high `op_rate`?

Reply:
That should work as long as the checkpoint `op_rate` is smaller than the truncate `op_rate`. What's the reason for not having a checkpoint here?

Reply:
I was wondering if we could simply replace it with the checkpoint operation provided by the framework but it sounds like it is not a good idea since we need timing.